gamemodes/horde/gamemode/perks/necromancer/necromancer_base.lua

```diff
@@ -8,70 +8,93 @@ Inflicts Frostbite buildup by {1} of base Cold damage. ({2} + {3} per level, up
 Uses Mind instead of Armor.
 Has access to spells for Void Projector.]]
 PERK.Icon = "materials/subclasses/necromancer.png"
+
 PERK.Params = {
-    [1] = {percent = true, base = 0.15, level = 0.01, max = 0.4, classname = "Necromancer"},
-    [2] = {value = 0.15, percent = true},
-    [3] = {value = 0.01, percent = true},
-    [4] = {value = 0.4, percent = true},
-    [5] = {percent = true, base = 0, level = 0.01, max = 0.25, classname = "Necromancer"},
-    [6] = {value = 0.01, percent = true},
-    [7] = {value = 0.25, percent = true},
+    [1] = { percent = true, base = 0.15, level = 0.01, max = 0.4, classname = "Necromancer" },
+    [2] = { value = 0.15, percent = true },
+    [3] = { value = 0.01, percent = true },
+    [4] = { value = 0.4, percent = true },
+    [5] = { percent = true, base = 0, level = 0.01, max = 0.25, classname = "Necromancer" },
+    [6] = { value = 0.01, percent = true },
+    [7] = { value = 0.25, percent = true },
 }
+
 PERK.Hooks = {}
 
-PERK.Hooks.Horde_OnPlayerDamageTaken = function (ply, dmginfo, bonus)
-    if not ply:Horde_GetPerk("necromancer_base")  then return end
-    if HORDE:IsColdDamage(dmginfo) then
-        bonus.resistance = bonus.resistance + ply:Horde_GetPerkLevelBonus("necromancer_base")
+PERK.Hooks.Horde_OnPlayerDamageTaken = function ( ply, dmginfo, bonus )
+    if !ply:Horde_GetPerk( "necromancer_base" )  then return end
```

Comment:
Please use Lua operators in this case this should be `if not ply:Horde_GetPerk( "necromancer_base" ) then return end`